pkg/console/controllers/service/controller.go

```diff
@@ -0,0 +1,170 @@
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	corev1Client coreclientv1.CoreV1Interface,
+	serviceInformer coreinformersv1.ServiceInformer,
+	// names
+	targetNamespace string,
+	serviceName string,
+	// events
+	recorder events.Recorder,
+) *ServiceSyncController {
+
+	corev1Client.Services(targetNamespace)
+
+	ctrl := &ServiceSyncController{
+		operatorConfigClient: operatorConfigClient,
+		serviceClient:        corev1Client,
+		// names
+		targetNamespace: targetNamespace,
+		serviceName:     serviceName,
+		// events
+		recorder:     recorder,
+		cachesToSync: nil,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	serviceInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		serviceInformer.Informer().HasSynced,
+	)
+
+	return ctrl
+}
+
+func (c *ServiceSyncController) sync() error {
+	startTime := time.Now()
+	klog.V(4).Infof("started syncing service %q (%v)", startTime)
+	defer klog.V(4).Infof("finished syncing service %q (%v)", time.Since(startTime))
+
+	operatorConfig, err := c.operatorConfigClient.Get(api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorsv1.Managed:
+		klog.V(4).Infoln("console is in an managed state: syncing service")
+	case operatorsv1.Unmanaged:
+		klog.V(4).Infoln("console is in an unmanaged state: skipping service sync")
+		return nil
+	case operatorsv1.Removed:
+		klog.V(4).Infoln("console is in an removed state: deleting service")
+		return c.removeService()
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	if !management.IsOperatorManaged(operatorConfig.Spec.ManagementState) {
+		return nil
+	}
+
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+	_, _, svcErr := c.SyncService()
+	status.HandleProgressingOrDegraded(updatedOperatorConfig, "ServiceSync", "FailedApply", svcErr)
+	status.SyncStatus(c.operatorConfigClient, updatedOperatorConfig)
+
+	return svcErr
+}
+
+func (c *ServiceSyncController) SyncService() (consoleService *corev1.Service, changed bool, err error) {
+	return resourceapply.ApplyService(c.serviceClient, c.recorder, service.DefaultService(nil))
+}
+
+func (c *ServiceSyncController) removeService() error {
+	klog.V(2).Info("deleting console service")
+	defer klog.V(2).Info("finished deleting console service")
+	return c.serviceClient.Services(c.targetNamespace).Delete(service.Stub().Name, &metav1.DeleteOptions{})
+}
+
+// boilerplate, since this controller is not making use of monis.app/go boilerplate
+func (c *ServiceSyncController) Run(workers int, stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	defer c.queue.ShutDown()
+	klog.Infof("starting %v", controllerName)
+	defer klog.Infof("shutting down %v", controllerName)
+	if !cache.WaitForCacheSync(stopCh, c.cachesToSync...) {
+		return
```

Comment:
lets log this cache error 
```suggestion
		klog.Infoln("caches did not sync")
		runtime.HandleError(fmt.Errorf("caches did not sync"))
		return
```

Reply:
Sounds good.